cli/src/parse.rs

```diff
@@ -39,11 +42,104 @@ impl fmt::Display for Stats {
     }
 }
 
+#[derive(Debug, Copy, Clone)]
+pub struct ParseTheme {
+    pub node_kind: Option<Color>,
+    pub node_text: Option<Color>,
+    pub field: Option<Color>,
+    pub token: Option<Color>,
+    pub row_color: Option<Color>,
+    pub row_color_named: Option<Color>,
+    pub extra: Option<Color>,
+    pub error: Option<Color>,
+}
+
+impl ParseTheme {
+    #[must_use]
+    pub const fn blank() -> Self {
+        Self {
+            node_kind: None,
+            node_text: None,
+            field: None,
+            token: None,
+            row_color: None,
+            row_color_named: None,
+            extra: None,
+            error: None,
+        }
+    }
+}
+
+impl Default for ParseTheme {
+    fn default() -> Self {
+        const GRAY: Color = Color::Rgb(RgbColor(118, 118, 118));
+        Self {
+            node_kind: Some(AnsiColor::BrightCyan.into()),
+            node_text: Some(GRAY),
+            field: Some(AnsiColor::Blue.into()),
+            token: Some(GRAY),
+            row_color: Some(AnsiColor::White.into()),
+            row_color_named: Some(AnsiColor::BrightYellow.into()),
+            extra: Some(AnsiColor::BrightMagenta.into()),
+            error: Some(AnsiColor::Red.into()),
+        }
+    }
+}
+
+#[derive(Debug, Copy, Clone, Deserialize, Serialize)]
+pub struct Rgb(pub u8, pub u8, pub u8);
+
+impl From<Rgb> for RgbColor {
+    fn from(val: Rgb) -> Self {
+        Self(val.0, val.1, val.2)
+    }
+}
+
+#[derive(Debug, Copy, Clone, Default, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct Config {
+    pub parse_theme: ParseThemeRaw,
+}
+
+#[derive(Debug, Copy, Clone, Default, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct ParseThemeRaw {
+    pub node_kind: Option<Rgb>,
+    pub node_text: Option<Rgb>,
+    pub field: Option<Rgb>,
+    pub token: Option<Rgb>,
+    pub row_color: Option<Rgb>,
+    pub row_color_named: Option<Rgb>,
+    pub extra: Option<Rgb>,
+    pub error: Option<Rgb>,
+}
+
+impl From<ParseThemeRaw> for ParseTheme {
+    fn from(value: ParseThemeRaw) -> Self {
+        let resolve_val = |val: Option<Rgb>, default: Option<Color>| -> Option<Color> {
+            val.map_or_else(|| default, |v| Some(Into::<RgbColor>::into(v).into()))
+        };
+        let default = Self::default();
+
+        Self {
+            node_kind: resolve_val(value.node_kind, default.node_kind),
+            node_text: resolve_val(value.node_text, default.node_text),
+            field: resolve_val(value.field, default.field),
+            token: resolve_val(value.token, default.token),
+            row_color: resolve_val(value.row_color, default.row_color),
+            row_color_named: resolve_val(value.row_color_named, default.row_color_named),
+            extra: resolve_val(value.extra, default.extra),
+            error: resolve_val(value.error, default.error),
+        }
```

Comment:
two nits - `val_or_default` is a better name imo, and the `Into` chain is a bit crazy

```suggestion
        let val_or_default = |val: Option<Rgb>, default: Option<Color>| -> Option<Color> {
            val.map_or(default, |v| Some(Color::Rgb(v.into())))
        };
        let default = Self::default();

        Self {
            node_kind: val_or_default(value.node_kind, default.node_kind),
            node_text: val_or_default(value.node_text, default.node_text),
            field: val_or_default(value.field, default.field),
            token: val_or_default(value.token, default.token),
            row_color: val_or_default(value.row_color, default.row_color),
            row_color_named: val_or_default(value.row_color_named, default.row_color_named),
            extra: val_or_default(value.extra, default.extra),
            error: val_or_default(value.error, default.error),
        }
```

Reply:
Yeah I definitely felt a little weird writing that many `into`s, much happier with this version